hera_qm/xrfi.py

```diff
@@ -222,6 +224,223 @@ def robust_divide(num, den):
+
+    # Recompute model with first pass weights
+    model, _, _ = dspec.fourier_filter(
+        freqs, data, wgts, filter_centers, filter_half_widths, mode=mode,
+        suppression_factors=suppression_factors, eigenval_cutoff=eigenval_cutoff,
+        cache=cache
+    )
+    res = data - model
+
+    # Use noise standard deviation given or estimate it from residuals
+    if noise is not None:
+        sigma = noise
+
+    else:
+        sigma = np.median(
+            np.abs(res - np.median(res, keepdims=True, axis=1)), axis=1, keepdims=True,
+        )
+        sigma /= 0.675
+
+    # Determine weights
+    wgts = np.where(np.abs(res) <= nsig * sigma, 1, 0)
+    return wgts
+
+
+def channel_diff_flagger(data, noise, nsig=10, kernel_widths=[1, 2, 4], wgts=None):
+    """
+    Flags auto-correlations using channel differencing kernels
+
+    Parameters:
+    ----------
+    data: np.ndarray
+        2D data array of the shape (time, frequency)
+    noise: np.ndarray
+        2D array for containing an estimate of the noise standard deviation. Must be the same shape as the data
+    nsig: float, default=10
+        The number of sigma in the metric above which to flag pixels.
+    kernel_width: list
+        Widths of the convolution kernels used to produce model
+    wgts: np.ndarray, default=None
+        2D array of real weights. Must be the same shape as data.
+
+    Returns:
+    -------
+    wgts: np.ndarray
+        Array containing weights for each antenna with values of either zero or one with
+        zero identifying channels found to have RFI
+    """
+    # If weights is None, set value to one
+    if wgts is None:
+        wgts = np.ones_like(data)
+
+    # Iterate through differencing kernels of different widths
+    for width in kernel_widths:
+        kernel = np.ones(2 * width + 1)
+        kernel_std = np.sqrt(1 + 1 / (2 * width))
+        dw = data * wgts
+        residual = np.zeros_like(data)
+
+        # For each time integration, identify outliers
+        for t in range(data.shape[0]):
+            _d = np.convolve(dw[t], kernel, mode="same") - dw[t]
+            _w = np.convolve(wgts[t], kernel, mode="same") - wgts[t]
+            residual[t] = dw[t] - _d / _w.clip(1, np.inf)
+            wgts[t] = np.where(residual[t] > noise[t] * kernel_std * nsig, 0, wgts[t])
+
+    return wgts
+
+
+auto_flaggers = {'dpss_flagger': dpss_flagger, 'channel_diff_flagger': channel_diff_flagger}
+
+
+def flag_autos(data, flag_method, nsig=10, int_count=None, antenna_class=None, wgts=None,
+               array_wgts_threshold=0.2, **flag_kwargs):
+    """
+    Driver function for identifying RFI in auto-correlations. Flags auto-correlations on a per antenna basis using
+    a given function and consolidates the weights into a single array-wide flag array.
+
+    Parameters:
+    ----------
+    data: DataContainer
+        DataContainer containing numpy arrays of auto-correlation data with shape (ntimes, nfreqs)
+    flag_method: str
```

Comment:
I'd suggest picking a default

Reply:
Picked the `channel_diff_flagger` as a default.